### PR TITLE
Prometheus: add ingress paths value

### PR DESCRIPTION
**Pull request by markvdw4**
To enable prometheus federation the 'federate' endpoint needs to be reachable on port 9090, to do this you need to add a new path to the server ingress template. In this PR I add a new value for server ingress paths and if it is set then we inject the desired paths into the server ingress template.

In order to parse properly paths must be defined in JSON:

```
server:
  ingress:
    paths: [{"backend": {"serviceName": "promfed","servicePort": 9090},"path": '/federate'}]
```

### CODE REVIEW

**Comment by markvdw4** [comment]
my github should be properly linked to my linuxfoundation/cncf account where I signed the CLA

File: stable/prometheus/README.md [hunk 1]
```diff
@@ -223,6 +224,20 @@ server:
...
+Prometheus federation requires the source server's 'federate' endpoint to be scraped on port 9090, to define the needed Ingress paths:
+
+```
+server:
+  ingress:
+    ## Prometheus server Ingress paths
+    ## Must be used for Prometheus federation
+    ##
+    paths: [{"backend": {"serviceName": "prometheusfederation", "servicePort": 9090}, "path": '/federate'}]
+```
```
Comment (unguiculus) [inline stable/prometheus/README.md:238]: I'd prefer this style:

```yaml
server:
  ingress:
    ## Prometheus server Ingress paths
    ## Must be used for Prometheus federation
    ##
    paths: 
      - path: /federate
        backend: 
          serviceName: prometheusfederation
          servicePort: 9090  
```

File: stable/prometheus/values.yaml [hunk 2]
```diff
@@ -320,6 +320,11 @@ server:
     hosts: []
     #   - prometheus.domain.com
 
+    ## Prometheus server Ingress paths
+    ## Must be used for Prometheus federation
+    ##
+    paths: []
```
Comment (unguiculus) [inline stable/prometheus/values.yaml:326]: I don't like the name. It is really something like `additionalPaths`.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by mgoodness** [comment]
The `/federate` endpoint is already exposed by the server Ingress. By default it's exposed on port 80, but that can be changed by setting `server.service.servicePort=9090`. Unless you absolutely *must* have different endpoints exposed on different ports, I think we should avoid the extra complexity this introduces.

**Review by mgoodness — CHANGES_REQUESTED** [review]
The /federate endpoint is already exposed by the server Ingress. By default it's exposed on port 80, but that can be changed by setting `server.service.servicePort=9090` (for example). Unless you absolutely *must* have different endpoints exposed on different ports, I think we should avoid the extra complexity this introduces.

**Comment by mgoodness** [comment]
Closing this. Feel free to re-open if you really do need separate endpoints/ports.

**Pull request closed without merging**